PyInstaller/depend/dylib.py

```diff
@@ -340,3 +340,59 @@ def match_func(pth):
             f.flush()
     except Exception:
         pass
+
+
+def mac_is_binary_signed(filename):
+    """
+    Check if the given macOS binary file is signed.
+    """
+    from macholib.MachO import MachO
+    from macholib import mach_o  # constants
+
+    # Open the file
+    try:
+        m = MachO(filename)
+    except Exception:
+        return False
+
+    # Walk over all headers and check if any contains LC_CODE_SIGNATURE
+    # load command
+    for header in m.headers:
+        for cmd in header.commands:
+            if cmd[0].cmd == mach_o.LC_CODE_SIGNATURE:
+                return True
+    return False
+
+
+def mac_strip_signature(libname, distname):
+    """
+    On macOS, strip away the signature from the binary file. As we may
+    not be collecting all components from a signed framework bundle, the
+    collection may invalidate the existing signature on a collected
+    shared library, which will prevent the latter from being loaded.
+    """
+    from ..compat import exec_command_rc
+
+    # For now, limit this only to Python shared library. Other shared
+    # library files from Python.framework bundle also seem to be signed,
+    # but their signature is not invalidated by partial collection like
+    # it is for Python library...
+    if os.path.basename(libname) != 'Python':
+        return
+    if not mac_is_binary_signed(libname):
+        return
+    # Run codesign --remove-signature libname
+    try:
+        logger.debug("Attempting to remove signature from %s", libname)
```

Comment:
```suggestion
        logger.debug("Removing signature from %s", libname)
```
We assume remving works. Or does it fail often?

Reply:
Good point. Fixed.